core/domain/stats_jobs_one_off_test.py

```diff
@@ -122,6 +123,48 @@ def test_creation_of_stats_model_for_v1(self):
             exploration_stats.state_stats_mapping[
                 'End'].useful_feedback_count_v1, 0)
 
+    def test_that_state_answers_sharded_models_accumulate_stats(self):
+        with self.swap(
+            stats_models.StateAnswersModel, '_MAX_ANSWER_LIST_BYTE_SIZE',
+            100000):
+
+            submitted_answer_list = [
+                stats_domain.SubmittedAnswer(
+                    'answer a', 'TextInput', 0, 1,
+                    exp_domain.EXPLICIT_CLASSIFICATION, {}, 'session_id_v',
+                    10.0),
+                stats_domain.SubmittedAnswer(
+                    'answer ccc', 'TextInput', 1, 1,
+                    exp_domain.DEFAULT_OUTCOME_CLASSIFICATION, {},
+                    'session_id_v', 3.0),
+                stats_domain.SubmittedAnswer(
+                    'answer bbbbb', 'TextInput', 1, 0,
+                    exp_domain.EXPLICIT_CLASSIFICATION, {}, 'session_id_v',
+                    7.5),
+            ]
+            stats_services.record_answers(
+                self.exp_id, self.exploration.version,
+                'Home', 'TextInput',
+                submitted_answer_list * 200)
+
+        job_id = stats_jobs_one_off.GenerateV1StatisticsJob.create_new()
+        stats_jobs_one_off.GenerateV1StatisticsJob.enqueue(job_id)
+
+        self.assertEqual(self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+        exploration_stats = stats_services.get_exploration_stats_by_id(
+            self.exp_id, self.exploration.version)
+
+        self.assertEqual(
+            exploration_stats.state_stats_mapping[
+                'Home'].total_answers_count_v1, 602)
```

Comment:
This seems slightly odd to me. I understand the 600 and the 400 but where do the extra 2 and 1 come from?

It's probably legit, but please add a comment to clarify.

Reply:
@seanlip Oh, the 2 is from the couple of answers in the setup method

Reply:
OK, please add a comment to clarify.

Reply:
Done.